designs/multiple-worker-node-groups.md

```diff
@@ -223,7 +223,11 @@ Right now, the cli assumes that there will be only one group and it treats worke
 
 Also, it needs to be made sure that at the least one of the worker node groups does not have `NoExecute` or `NoSchedule` taint. This validation will be done at the preflight validation stage.
 
-To delete a worker node group, we will first drain the worker node group and then set the replica count for that group to 0 in CAPI spec file. Upon applying the CAPI spec on the workload cluster, that node group will be deleted.
+To delete a worker node group we will perform the following steps.
+
+* We will add a name field in the cluster spec, so that a user can specify names of each group.
```

Comment:
Can we add something about what we will do with existing clusters? Since this is an API change
I think we already talked about it: for existing clusters, we will assign a default name to the first one (we only support one so there can't me clusters with 2 or more), like `worker-node-group-0`. This can be implemented in both the cli and the mutation webhook.